File: Gateway/src/main/java/com/quizGpt/Gateway/Quiz/Dto/CreateQuizRequestDto.java
package com.quizGpt.Gateway.Quiz.Dto;

import org.hibernate.annotations.GenericGenerator;
import org.springframework.data.annotation.Id;

import com.fasterxml.jackson.annotation.JsonProperty;

import jakarta.persistence.GeneratedValue;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;


@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class CreateQuizRequestDto {
    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "org.hibernate.id.UUIDGenerator")
    private String id;

    @NotBlank
    @JsonProperty("topic")
    private String topic;

    @NotBlank
    @JsonProperty("numberOfQuestions")
    private Long numberOfQuestions;

    @NotBlank
    @JsonProperty("numberOfOptionsPerQuestion")
    private Long numberOfOptionsPerQuestion;

    @NotBlank
    @JsonProperty("difficulty")
    private String difficulty;
}

File: graphql-version/resource-server/src/dtos/createQuizRequestDto.ts
import { Long } from "typeorm";
import { v4 as uuidv4 } from "uuid";

export class CreateQuizRequestDto {
  id: string;
  userId: string;
  title: String;
  topic: String;
  numberOfQuestions: Long;
  numberOfOptionsPerQuestion: Long;
  difficulty: String;

  constructor(
    userId: string,
    topic: String,
    title: String,
    numberOfQuestions: Long,
    numberOfOptionsPerQuestion: Long,
    difficulty: String
  ) {
    this.id = uuidv4();
    this.userId = userId;
    this.title = title;
    this.topic = topic;
    this.numberOfQuestions = numberOfQuestions;
    this.numberOfOptionsPerQuestion = numberOfOptionsPerQuestion;
    this.difficulty = difficulty;
  }
}
